source_pubsub/client.go

```diff
@@ -118,25 +130,28 @@ func NewGRPCClient(ctx context.Context, config Config, sdkPos sdk.Position) (*Pu
 		replayPreset = proto.ReplayPreset_EARLIEST
 	}
 
+	sdkPos.SetTopics(config.TopicNames)
+
 	return &PubSubClient{
 		conn:         conn,
 		pubSubClient: proto.NewPubSubClient(conn),
 		codecCache:   make(map[string]*goavro.Codec),
 		unionFields:  make(map[string]map[string]struct{}),
-		currReplayID: sdkPos,
 		replayPreset: replayPreset,
 		buffer:       make(chan sdk.Record),
 		ticker:       time.NewTicker(config.PollingPeriod),
-		topicName:    config.TopicName,
+		topicNames:   config.TopicNames,
+		sdkPos:       sdkPos,
```

Comment:
I'd call this something like currentPos or something similar that represent what it is.